R/egib_layer_download.R

```diff
@@ -0,0 +1,108 @@
+#' @title Download Land and Building Register (EGiB) layers
+#'
+#' @param county County name in Polish. Check [`county_names()`] function.
+#' @param TERYT County ID (4 characters).
+#' @param layer Requested layer, `dzialki` (parcels)default. Other common layer is `budynki` (buildings).
+#' You can check available layers by [`egib_layers()`] data set.
+#' @param outdir (Optional) name of the output directory where the data has to be stored;
+#' NULL by default.
+#' @param ... any other parameter passed to [`sf::st_read()`] function
+#'
+#' @return Simple feature data frame of objects for requested layer.
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' parcels <- egib_layer_download(county = "wołowski", layer = "dzialki")
+#' buildings <- egib_layer_download(county = "milicki", layer = "budynki", outdir = ".")
+#' }
+#'
+#' @references
+#' The EGiB data (Ewidencja Gruntów i Budynków) consist of 2 primary layers:
+#' cadastral data (often called dzialki) and buildings footprints (budynki).
+#' The data is maintained on county level, which results in 380 units/different
+#' sources of data. It may contain additional layers like points of detailed
+#' horizontal and vertical geodetic control network (osnowa_pozioma
+#' and osnowa_pionowa respectively).
+#'
+#' \url{https://www.geoportal.gov.pl/en/data/land-and-building-register-egib/}
+#' \url{https://www.geoportal.gov.pl/en/data/detailed-control-network-database-bdsog/}
+#'
+
+
+# county <- "trzebnicki"
+# # # county <- "Łomża"
+# TERYT <- NULL
+# layer <- "Osnowa"
+# outdir <- "~/Downloads/"
+
+egib_layer_download <- function(county = NULL, TERYT = NULL, layer = "dzialki", outdir = NULL, ...) {
```

Comment:
Could the argument `layer` take English names, i.e. `parcels` and `buildings`?

Reply:
I think we can simplify the function name to `egib_` (`download` or `get`; see comment below).